docs/start/getting-started/fragments/android/integrate.md

```diff
@@ -2,44 +2,98 @@ Next you'll use the generated model to read and write data. In this section you'
 
-First, we'll add the DataStore plugin and configure Amplify. Typically, a good place to do this is in the [`onCreate()`](https://developer.android.com/reference/android/app/Application#onCreate()) method of [Android's `Application` class](https://developer.android.com/reference/android/app/Application). For the purposes of this tutorial, you can place these examples in the `onCreate()` method of the `MainActivity` class.
+First, we'll add the DataStore plugin and configure Amplify by creating an Application class and overriding the `onCreate()` method.
 
-1. Open `MainActivity` and add the following code to the bottom of the `onCreate()` method:
+1. Go to **File** > **New** and select either **Java Class** or **Kotlin File/Class**.
+
+1. Select **Class**.
+
+1. Type **MyAmplifyApplication** in the **Name** field.
+
+1. Paste the following code to initialize Amplify:
 
   <amplify-block-switcher>
   <amplify-block name="Java">
   
-  ```java
-  try {
-      Amplify.addPlugin(new AWSDataStorePlugin());
-      Amplify.addPlugin(new AWSApiPlugin());
-      Amplify.configure(getApplicationContext());
-
-      Log.i("Tutorial", "Initialized Amplify");
-  } catch (AmplifyException e) {
-      Log.e("Tutorial", "Could not initialize Amplify", e);
-  }
-  ```
+    ```java
+    public class MyAmplifyApplication extends Application {
+        public void onCreate() {
+            super.onCreate();
+            try {
+                Amplify.addPlugin(new AWSApiPlugin());
+                Amplify.addPlugin(new AWSDataStorePlugin());
+                Amplify.configure(getApplicationContext());
+
+                Log.i("Tutorial", "Initialized Amplify");
+            } catch (AmplifyException e) {
+                Log.e("Tutorial", "Could not initialize Amplify", e);
+            }
+        }
+    }
+    ```
 
   </amplify-block>
 
   <amplify-block name="Kotlin">
 
   ```kotlin
-  try {
-      Amplify.addPlugin(AWSDataStorePlugin())
-      Amplify.addPlugin(AWSApiPlugin())
-      Amplify.configure(applicationContext)
-
-      Log.i("Tutorial", "Initialized Amplify")
-  } catch (e: AmplifyException) {
-      Log.e("Tutorial", "Could not initialize Amplify", e)
+  class MyAmplifyApplication : Application() {
+      override fun onCreate() {
+          super.onCreate()
+          try {
+              Amplify.addPlugin(AWSApiPlugin())
+              Amplify.addPlugin(AWSDataStorePlugin())
+              Amplify.configure(applicationContext)
+              Log.i("Tutorial", "Initialized Amplify")
+          } catch (e: AmplifyException) {
+              Log.e("Tutorial", "Could not initialize Amplify", e)
+          }
+      }
   }
   ```
 
   </amplify-block>
   </amplify-block-switcher>
 
+1. Open **manifests** > **AndroidManifest.xml** to configure your application.
+
+1. Add the `android:name` and `tools:replace` attributes to the `application` node:
+
+  ```xml
+  <application
+      android:name=".MyAmplifyApplication"
+      tools:replace="android:name"
```

Comment:
This probably breaks Multidex on devices with API < 21.

The Google docs on this topic propose [three ways to embed multidex into the application class](https://developer.android.com/studio/build/multidex). Our library currently uses the first of these options. More resilient for consumers would probably be either the second or third options. The forth (and even better option) would be to raise the `minSdk` on our product tin to 21, so that the on-boarding were as smooth as possible.

Reply:
Gotcha, yeah, skimmed there earlier. I think we need to huddle up on this Multidex stuff and think about the right approach.